### DISCUSSION TITLE
Showing Chatter (aka FeedItem)

### DISCUSSION THREAD

**Discussion by mayday6971** [deejay-hub/timeline-lwc#218] [Q&A] [answered]
Hello! I have to say this is a great idea for a component. I was curious how to setup Chatter? I tried the basics and for whatever reason I cannot get the linkage to work.

**Comment by deejay-hub** [comment, marked as answer]
Chatter isn't supported in the current code base.

Adding support for FeedItem would require some careful thought (polls vs post, images, large data blocks). Something I will leave up to the community. There are 2 options
1. Someone sends a pull request with the necessary code changes
2. You fork the repo and add the changes

In the Timeline Service (getChildObjects method) you would have to play around with the checks performed
 ```
List<Schema.ChildRelationship> objectRelationships = ((SObject) (Type.forName('Schema.' + parentObject)
                .newInstance()))
            .getSObjectType()
            .getDescribe()
            .getChildRelationships();
            
        for (Schema.Childrelationship eachRelationship : objectRelationships) {

            if (
                eachRelationship.getChildSObject().getDescribe().isAccessible() &&
                !eachRelationship.getChildSObject().getDescribe().getLabel().contains('Histories') &&
                **eachRelationship.getRelationshipName() != null**
            ) {
```
Technically there is a relationship via Feeds for most objects. Trying to use it though would result in it being ignored since the relationshipName isn't returned in the getDescribe call. So in the code above we filter it out and don't consider it valid.

You could add logic to add the relationship name yourself. I don't plan on adding it given Slack's strategic appearance but if someone sends me working code then I'll happily incorporate it.

**Reply by deejay-hub** [reply]
Updated the FAQ doc clarifying that chatter posts are not supported.
https://quip.com/6yvoAcBukqZB

**Comment by mayday6971** [comment]
I was thinking the FeedItem was best but I do think you have a point to use CaseFeed. As for displaying the images, I would generally just let the Body render, as last I checked it can and does. I'll play around and see where I can get.

Thanks for the response and for updating the documentation.
